metasynth/spec.py

```diff
@@ -0,0 +1,149 @@
+from typing import Optional, Union
+
+import polars as pl
+
+from metasynth.provider import BaseDistribution
+from metasynth.privacy import BasePrivacy
+
+AVAILABLE_TYPES = {
+    "distribution": [BaseDistribution, type(BaseDistribution), str],
+    "unique": [bool],
+    "description": [str],
+    "privacy": [BasePrivacy, type(BasePrivacy), str],
+    "prop_missing": [float],
+    "fit_kwargs": [dict]
+}
+
+
+def type_check(key, value, type_dict=AVAILABLE_TYPES):
+    if not any(isinstance(value, allowed_type) for allowed_type in
```

Comment:
You should be able to use `isinstance(value, type_dict[key])` to check for multiple types at once.